### PR TITLE
fixed compatability with new LibGDX

### PR DESCRIPTION
**Pull request by EtK2000**
changed return type to SnapshotArray<> as apposed to Array<>

### CODE REVIEW

**Comment by EtK2000** [comment]
Stating that it took my newest pull, I'll change this to the appropriate name.

**Comment by intrigus** [comment]
It's absolutely impossible to see what you changed.

**Comment by EtK2000** [comment]
I know, I kinda rewrote the whole thing.
I rewrote it in Android Studio, I presume it formatted some things.

**Comment by yangweigbh** [comment]
hi EtK2000, thanks for your commit, can you reformat your code and submit again, i can not figure out which part has changed.

**Comment by EtK2000** [comment]
Sure, I'll fix it later today.
Btw, I think this breaks comparability with older LibGDX in favor of new versions.
If so, I'll check which version changed the API.

**Comment by EtK2000** [comment]
I have pushed my commit after formatting, now the changes are readable.
I also moved some things around to keep the order consistent between this and LibGDX.

File: libgdx-cardboard-core/src/com/badlogic/gdx/backends/android/CardBoardGraphics.java [hunk 1]
```diff
@@ -197,22 +174,45 @@ public int getWidth () {
...
+
    /** This instantiates the GL10, GL11 and GL20 instances. Includes the check for certain devices that pretend to support GL11 but
     * fuck up vertex buffer objects. This includes the pixelflinger which segfaults when buffers are deleted as well as the
-    * Motorola CLIQ and the Samsung Behold II.
-    *
-    * @param gl */
+    * Motorola CLIQ and the Samsung Behold II. */
    private void setupGL () {
-      if (gl20 != null) return;
+      if (config.useGL30 && glVersion.getMajorVersion() > 2) {
```
Comment (yangweigbh) [inline libgdx-cardboard-core/src/com/badlogic/gdx/backends/android/CardBoardGraphics.java]: glversion is null here

**Comment by EtK2000** [comment]
Fixed it, thanks for pointing it out.

**Review by yangweigbh — APPROVED** [review]

**Comment by EtK2000** [comment]
Don't forget to close issue 9.
https://github.com/yangweigbh/Libgdx-CardBoard-Extension/issues/9

**Pull request merged**